adal/src/androidTest/java/com/microsoft/aad/adal/AcquireTokenRequestTest.java

```diff
@@ -873,6 +873,54 @@ public void testVerifyManifestPermissionMissingMultiPermissions() throws Interru
         }
     }
 
+    public void testSilentRequestMissingPermissionHandlingForAndroid22andBelow() throws InterruptedException, PackageManager.NameNotFoundException {
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+            final FileMockContext mockContext = createMockContext();
+            when(mockContext.getPackageManager().checkPermission(Mockito.refEq("android.permission.GET_ACCOUNTS"),
+                    Mockito.anyString())).thenReturn(PackageManager.PERMISSION_GRANTED);
+            when(mockContext.getPackageManager().checkPermission(Mockito.refEq("android.permission.MANAGE_ACCOUNTS"),
+                    Mockito.anyString())).thenReturn(PackageManager.PERMISSION_DENIED);
+            when(mockContext.getPackageManager().checkPermission(Mockito.refEq("android.permission.USE_CREDENTIALS"),
+                    Mockito.anyString())).thenReturn(PackageManager.PERMISSION_DENIED);
+
+            AuthenticationSettings.INSTANCE.setUseBroker(true);
+            final AuthenticationContext authContext = new AuthenticationContext(mockContext,
+                    VALID_AUTHORITY, false);
+            final TestAuthCallback callback = new TestAuthCallback();
+            authContext.acquireTokenSilentAsync("resource", "clientid", "userid", callback);
+            final CountDownLatch signal = new CountDownLatch(1);
+            signal.await(ACTIVITY_TIME_OUT, TimeUnit.MILLISECONDS);
+
+            assertNotNull(callback.getCallbackException());
+            assertTrue(callback.getCallbackException() instanceof UsageAuthenticationException);
+            final UsageAuthenticationException usageAuthenticationException
+                    = (UsageAuthenticationException) callback.getCallbackException();
+            assertTrue(usageAuthenticationException.getMessage().contains("MANAGE_ACCOUNTS"));
+            assertTrue(usageAuthenticationException.getMessage().contains("USE_CREDENTIALS"));
+            assertEquals(ADALError.DEVELOPER_BROKER_PERMISSIONS_MISSING, usageAuthenticationException.getCode());
+        }
+    }
+
+    public void testSilentRequestMissingPermissionHandlingForAndroid23andAbove() throws InterruptedException, PackageManager.NameNotFoundException {
+        final FileMockContext mockContext = createMockContext();
+        when(mockContext.getPackageManager().checkPermission(Mockito.refEq("android.permission.GET_ACCOUNTS"),
+                Mockito.anyString())).thenReturn(PackageManager.PERMISSION_DENIED);
+        AuthenticationSettings.INSTANCE.setUseBroker(true);
+        final AuthenticationContext authContext = new AuthenticationContext(mockContext,
+                VALID_AUTHORITY, false);
+        try {
+            authContext.acquireTokenSilentSync("resource", "clientid", "userid");
+            fail("Expect an exception");
+        } catch (final Exception exception) {
+            assertTrue(exception instanceof AuthenticationException);
```

Comment:
>assertTrue(exception instanceof AuthenticationException); [](start = 12, length = 57)

you don't need to do this, catch AuthenticationException, let the others fall through